install-ci-tools.sh

```diff
@@ -200,7 +199,44 @@ EOF
     ranch)
       if [ ! -x ranch ] || [ "$(./ranch version)" != "$version" ]; then
         rm -f ranch
-        curl -fsSL http://ranch-updates.goodeggs.com/stable/ranch/${version}/linux-amd64.gz | gunzip > ranch
+        curl -fsSL "http://ranch-updates.goodeggs.com/stable/ranch/${version}/linux-amd64.gz" | gunzip > ranch_real
+        chmod +x ranch_real
+      cat > ranch <<"EOF"
+#!/bin/bash
+
+set -eo pipefail
+
+# Make sure and clean up
+trap "exit" INT TERM ERR
+trap "kill 0" EXIT
+
+
+script_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+if [ -z "$RANCH_PROXY_SSH_KEY" ]
+then
+  echo "This will break in the near future, please set RANCH_PROXY_SSH_KEY"
+  exec $script_dir/ranch_real "$@"
+fi
+
+echo "$RANCH_PROXY_SSH_KEY" | base64 -d > .ssh_key
+chmod 600 .ssh_key
```

Comment:
there's a security race here... create the file with 600 first, then cat into it?